src/NSwag.CodeGeneration/Models/ParameterModelBase.cs

```diff
@@ -44,6 +45,12 @@ public abstract class ParameterModelBase
             Type = typeName;
             Name = parameterName;
             VariableName = variableName;
+
+            var propertyNameGenerator = settings?.PropertyNameGenerator ?? new DefaultPropertyNameGenerator();

```

Comment:
Is this (and DefaultPropertyNameGenerator.cs) really needed? Isnt a generator already registered/set by default?

Reply:
Yes, I think you can remove DefaultPropertyNameGenerator as there is already one for CSharp and TypeScript which is always set: 

https://github.com/RSuter/NJsonSchema/blob/master/src/NJsonSchema.CodeGeneration.CSharp/CSharpGeneratorSettings.cs#L41

Just throw an exception here, e.g.: 

    settings?.PropertyNameGenerator ?? throw new InvalidOperationException("PropertyNameGenerator not set.");